src/sbd-md.c

```diff
@@ -1072,24 +1057,21 @@ int servant(const char *diskname, int mode, const void* argp)
 	/* FIXME: check error */
 	sigprocmask(SIG_SETMASK, &servant_masks, NULL);
 
-	atexit(servant_exit);
-	servant_inform_parent = 1;
-
 	st = open_device(diskname, LOG_WARNING);
 	if (!st) {
-		return -1;
+		exit(EXIT_MD_IO_FAIL);
```

Comment:
Have to think over it. What I tried to do up to now is to mimic the previous behavior just via exit code instead of the potentially racy approach of sending a signal first and right afterwards ending the process, where - at least according to the documentation found - it can't be assured that the RT-signal arrives before the SIGCHLD which removes the pid from entry and prevents this from being found when the signal afterwards finally arrives.
Creating the signal was previously controlled via a variable and the return you found was before this being set.